### DISCUSSION TITLE
Runtime Config: internal reload vs external reload

### DISCUSSION THREAD

**Discussion by SunSparc** [grafana/mimir#4482] [Help and support] [answered]
I have Mimir setup to use GCS. The credentials are stored in a local file, and the location of that file is stored in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable, which Mimir uses by default for GCS. The credentials are automatically rotated periodically (eg once a day). The tool that I have rotating the credentials in that file can also restart Mimir, or send it a kill signal, or whatever else I need it to do.

I would like to know what the best practice is for restarting/reloading Mimir. It does not seem to automatically re-read the credentials file I have, so I assume it will need to be reloaded/restarted either internally (runtime config) or externally (restart, or kill signal)

I was reading [About Grafana Mimir runtime configuration](https://grafana.com/docs/mimir/latest/operators-guide/configure/about-runtime-configuration/), which has a mechanism for reloading Mimir. Using this feature requires having some configurations in a separate file.

Question 1: Can the `common` block go in a runtime config file? Like this?
```
mimir-config.yaml
---
runtime_config:
  file: "common.yaml"

====

common.yaml
---
common:
  storage:
    backend: gcs
    gcs:
      bucket_name: my-unique-bucket-name
```

Question 2: Will Mimir re-read the credentials file if I use the runtime config in question 1?

Question 3: What is the best practice for having Mimir re-read a credentials file?

**Comment by pstibrany** [comment, marked as answer]
> Question 1: Can the `common` block go in a runtime config file? Like this?

No, runtime config only supports very few fields (ingester instance limits, and per tenant limit `overrides`). Other fields are not supported in runtime config and will result in Mimir complaining about unknown fields.



> Question 3: What is the best practice for having Mimir re-read a credentials file?

Right now there is no way to ask Mimir to reload credentials files, the only way is to restart Mimir.

**Reply by SunSparc** [reply]
Thank you for the information. I will proceed with the restart approach.
